Okta.Core.Automation/Users/DeleteOktaUser.cs

```diff
@@ -0,0 +1,103 @@
+using Okta.Core.Models;
+
+namespace Okta.Core.Automation
+{
+    [Cmdlet("Delete", "OktaUser")]
+    public class DeleteOktaUser : OktaCmdlet
+    {
+        [Parameter(
+            Mandatory = false,
+            ValueFromPipelineByPropertyName = true,
+            Position = 0,
+            HelpMessage = "ID or username of the user to delete"
+        )]
+        public string Id { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            ValueFromPipelineByPropertyName = true,
+            Position = 1,
+            HelpMessage = "Search query string (see http://developer.okta.com/docs/api/resources/users.html#find-users)"
+        )]
+        public string Query { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            ValueFromPipelineByPropertyName = true,
+            Position = 2,
+            HelpMessage = "Use a filter (see http://developer.okta.com/docs/api/resources/users.html#list-users-with-a-filter)"
+        )]
+        public string Filter { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            ValueFromPipelineByPropertyName = true,
+            Position = 3,
+            HelpMessage = "Use advanced search (see http://developer.okta.com/docs/api/resources/users.html#list-users-with-search)"
+        )]
+        public bool Search { get; set; }
+
+        protected override void ProcessRecord()
+        {
+            var usersClient = Client.GetUsersClient();
+            try
+            {
+                List<User> users = new List<User>();
+                if (!string.IsNullOrEmpty(Id))
+                {
+                    var user = usersClient.Get(Id);
+                    if (user != null)
+                        users.Add(user);
+                }
+                else
+                {
+                    SearchType searchType = SearchType.Filter;
+                    if (Search)
+                    {
+                        searchType = SearchType.ElasticSearch;
+                    }
+                    var usersEnum = usersClient.GetFilteredEnumerator(query: Query, searchType: searchType, filter: new FilterBuilder(Filter));
+                    users = usersEnum.ToList<User>();
+                }
+
+                foreach (User user in users)
+                {
+                    string strUserLogin = user.Profile.Login;
+                    bool bContinue = false;
+                    try
+                    {
+                        usersClient.Deactivate(user);
+                        bContinue = true;
+                    }
+                    catch (OktaException oex2)
+                    {
+                        if (oex2.ErrorCode == OktaErrorCodes.ResourceNotFoundException)
+                        {
+                            bContinue = true;
+                        }
+                    }
+
+                    if (bContinue)
+                    {
+                        usersClient.Delete(user);
```

Comment:
Note that I tested it worked with a real list of users (filtered by username - they all started with "testuser")